Python 3.5.2 |Anaconda 4.2.0 (64-bit)| (default, Jul  2 2016, 17:53:06) 
[GCC 4.4.7 20120313 (Red Hat 4.4.7-1)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> my_list=[]
>>> my_list,id(my_list),type(my_list)
([], 139958683077448, <class 'list'>)
>>> my_list=[1,2,3,[1,2,3],(1,2,3),{1:"one",2:"two",3:"Three"}]
>>> my_list
[1, 2, 3, [1, 2, 3], (1, 2, 3), {1: 'one', 2: 'two', 3: 'Three'}]
>>> mylist[3]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'mylist' is not defined
>>> my_list[3]
[1, 2, 3]
>>> my_list[4]
(1, 2, 3)
>>> my_list[5]
{1: 'one', 2: 'two', 3: 'Three'}
>>> my_list[3,1]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: list indices must be integers or slices, not tuple
>>> my_list[3][0]
1
>>> my_list[3][1]
2
>>> my_list[3][2]
3
>>> my_list[4][0]
1
>>> my_list[4][1]
2
>>> my_list[4][2]
3
>>> my_list[1][0]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'int' object is not subscriptable
>>> my_list[5][1]
'one'
>>> my_list[5][2]
'two'
>>> my_list[5][3]
'Three'
>>> my_list[5][-1]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
KeyError: -1
>>> my_list[-1]
{1: 'one', 2: 'two', 3: 'Three'}
>>> my_list[-2]
(1, 2, 3)
>>> my_list[-3]
[1, 2, 3]
>>> my_list[-4]
3
>>> my_list[-5]
2
>>> my_list[-6]
1
>>> my_list[-1][1]
'one'
>>> my_list[-1][2]
'two'
>>> my_list[-1][3]
'Three'
>>> my_list[-2][0]
1
>>> my_list[-2][1]
2
>>> my_list[-2][3]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
IndexError: tuple index out of range
>>> my_list[-2][2]
3
>>> my_list[-3][0]
1
>>> my_list[-3][1]
2
>>> my_list[-3][2]
3
>>> my_list[-3][-1]
3
>>> my_list[-3][-2]
2
>>> my_list[-3][-3]
1
>>> my_list[::-1]
[{1: 'one', 2: 'two', 3: 'Three'}, (1, 2, 3), [1, 2, 3], 3, 2, 1]
>>> 1 in my_list
True
>>> 1 not in my_list
False
>>> max(my_list)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unorderable types: list() > int()
>>> l=[1,2,3,4,5,6]
>>> max(l)
6
>>> min(l)
1
>>> del my_list[5]
>>> my_list
[1, 2, 3, [1, 2, 3], (1, 2, 3)]
>>> my_list[4]
(1, 2, 3)
>>> my_list
[1, 2, 3, [1, 2, 3], (1, 2, 3)]
>>> del my_list[4]
>>> my_list
[1, 2, 3, [1, 2, 3]]
>>> del my_list[3]
>>> my_list
[1, 2, 3]
>>> del my_list[2]
>>> my_list
[1, 2]
>>> del my_list
>>> my_list
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'my_list' is not defined
>>> my_list=[1,2,3,[1,2,3],(1,2,3),{1:"one",2:"two",3:"Three"}]
>>> my_list
[1, 2, 3, [1, 2, 3], (1, 2, 3), {1: 'one', 2: 'two', 3: 'Three'}]
>>> my_list.count()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: count() takes exactly one argument (0 given)
>>> count(my_list)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'count' is not defined
>>> list.count(1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: descriptor 'count' requires a 'list' object but received a 'int'
>>> my_list.count()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: count() takes exactly one argument (0 given)
>>> my_list.count(1)
1
>>> my_list
[1, 2, 3, [1, 2, 3], (1, 2, 3), {1: 'one', 2: 'two', 3: 'Three'}]
>>> my_list.insert(3,1)
>>> my_list
[1, 2, 3, 1, [1, 2, 3], (1, 2, 3), {1: 'one', 2: 'two', 3: 'Three'}]
>>> my_list.count(1)
2
>>> l
[1, 2, 3, 4, 5, 6]
>>> my_list.extend(l)
>>> my_list
[1, 2, 3, 1, [1, 2, 3], (1, 2, 3), {1: 'one', 2: 'two', 3: 'Three'}, 1, 2, 3, 4, 5, 6]
>>> l
[1, 2, 3, 4, 5, 6]
>>> my_list.count(1)
3
>>> list.index(1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: descriptor 'index' requires a 'list' object but received a 'int'
>>> my_list.index(1)
0
>>> my_list.sort()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unorderable types: list() < int()
>>> l
[1, 2, 3, 4, 5, 6]
>>> l.sort()
>>> l.append(3)
>>> l.append(4)
>>> l.append(5)
>>> l
[1, 2, 3, 4, 5, 6, 3, 4, 5]
>>> l.sort()
>>> l
[1, 2, 3, 3, 4, 4, 5, 5, 6]
>>> my_list[0:]
[1, 1, 2, 3, [1, 2, 3], (1, 2, 3), {1: 'one', 2: 'two', 3: 'Three'}, 1, 2, 3, 4, 5, 6]
>>> my_list[:]
[1, 1, 2, 3, [1, 2, 3], (1, 2, 3), {1: 'one', 2: 'two', 3: 'Three'}, 1, 2, 3, 4, 5, 6]
>>> my_list[0:]
[1, 1, 2, 3, [1, 2, 3], (1, 2, 3), {1: 'one', 2: 'two', 3: 'Three'}, 1, 2, 3, 4, 5, 6]
>>> my_list[1:]
[1, 2, 3, [1, 2, 3], (1, 2, 3), {1: 'one', 2: 'two', 3: 'Three'}, 1, 2, 3, 4, 5, 6]
>>> my_list[4:]
[[1, 2, 3], (1, 2, 3), {1: 'one', 2: 'two', 3: 'Three'}, 1, 2, 3, 4, 5, 6]
>>> my_list[5:]
[(1, 2, 3), {1: 'one', 2: 'two', 3: 'Three'}, 1, 2, 3, 4, 5, 6]
>>> my_list[6:]
[{1: 'one', 2: 'two', 3: 'Three'}, 1, 2, 3, 4, 5, 6]
>>> my_list[7:]
[1, 2, 3, 4, 5, 6]
>>> my_list[10:]
[4, 5, 6]
>>> my_list[13:]
[]
>>> my_list[14:]
[]
>>> my_list
[1, 1, 2, 3, [1, 2, 3], (1, 2, 3), {1: 'one', 2: 'two', 3: 'Three'}, 1, 2, 3, 4, 5, 6]
>>> my_list[:1]
[1]
>>> my_list[:2]
[1, 1]
>>> my_list[:3]
[1, 1, 2]
>>> my_list[:4]
[1, 1, 2, 3]
>>> my_list[:5]
[1, 1, 2, 3, [1, 2, 3]]
>>> my_list[:6]
[1, 1, 2, 3, [1, 2, 3], (1, 2, 3)]
>>> my_list[:7]
[1, 1, 2, 3, [1, 2, 3], (1, 2, 3), {1: 'one', 2: 'two', 3: 'Three'}]
>>> my_list[:12]
[1, 1, 2, 3, [1, 2, 3], (1, 2, 3), {1: 'one', 2: 'two', 3: 'Three'}, 1, 2, 3, 4, 5]
>>> my_list[:13]
[1, 1, 2, 3, [1, 2, 3], (1, 2, 3), {1: 'one', 2: 'two', 3: 'Three'}, 1, 2, 3, 4, 5, 6]
>>> my_list[:14]
[1, 1, 2, 3, [1, 2, 3], (1, 2, 3), {1: 'one', 2: 'two', 3: 'Three'}, 1, 2, 3, 4, 5, 6]
>>> my_list[:15]
[1, 1, 2, 3, [1, 2, 3], (1, 2, 3), {1: 'one', 2: 'two', 3: 'Three'}, 1, 2, 3, 4, 5, 6]
>>> my_list[0:5]
[1, 1, 2, 3, [1, 2, 3]]
>>> my_list[0:7]
[1, 1, 2, 3, [1, 2, 3], (1, 2, 3), {1: 'one', 2: 'two', 3: 'Three'}]
>>> my_list[0:8]
[1, 1, 2, 3, [1, 2, 3], (1, 2, 3), {1: 'one', 2: 'two', 3: 'Three'}, 1]
>>> my_list[0:13]
[1, 1, 2, 3, [1, 2, 3], (1, 2, 3), {1: 'one', 2: 'two', 3: 'Three'}, 1, 2, 3, 4, 5, 6]
>>> my_list[4:10]
[[1, 2, 3], (1, 2, 3), {1: 'one', 2: 'two', 3: 'Three'}, 1, 2, 3]
>>> my_list[4:2]
[]
>>> my_list[:-1]
[1, 1, 2, 3, [1, 2, 3], (1, 2, 3), {1: 'one', 2: 'two', 3: 'Three'}, 1, 2, 3, 4, 5]
>>> my_list[:-13]
[]
>>> my_list[::-1]
[6, 5, 4, 3, 2, 1, {1: 'one', 2: 'two', 3: 'Three'}, (1, 2, 3), [1, 2, 3], 3, 2, 1, 1]
>>> my_list[:5:-1]
[6, 5, 4, 3, 2, 1, {1: 'one', 2: 'two', 3: 'Three'}]
>>> my_list[1:5:-1]
[]
>>> my_list[:10:-1]
[6, 5]
>>> my_list
[1, 1, 2, 3, [1, 2, 3], (1, 2, 3), {1: 'one', 2: 'two', 3: 'Three'}, 1, 2, 3, 4, 5, 6]
>>> my_list[:9:-1]
[6, 5, 4]
>>> my_list[:6:-1]
[6, 5, 4, 3, 2, 1]
>>> my_list[::-1]
[6, 5, 4, 3, 2, 1, {1: 'one', 2: 'two', 3: 'Three'}, (1, 2, 3), [1, 2, 3], 3, 2, 1, 1]
>>> my_list.reverse()
>>> my_list
[6, 5, 4, 3, 2, 1, {1: 'one', 2: 'two', 3: 'Three'}, (1, 2, 3), [1, 2, 3], 3, 2, 1, 1]
>>> my_list
[6, 5, 4, 3, 2, 1, {1: 'one', 2: 'two', 3: 'Three'}, (1, 2, 3), [1, 2, 3], 3, 2, 1, 1]
>>> my_list.reverse()
>>> my_list
[1, 1, 2, 3, [1, 2, 3], (1, 2, 3), {1: 'one', 2: 'two', 3: 'Three'}, 1, 2, 3, 4, 5, 6]
>>> my_list[::-1]
[6, 5, 4, 3, 2, 1, {1: 'one', 2: 'two', 3: 'Three'}, (1, 2, 3), [1, 2, 3], 3, 2, 1, 1]
>>> my_list
[1, 1, 2, 3, [1, 2, 3], (1, 2, 3), {1: 'one', 2: 'two', 3: 'Three'}, 1, 2, 3, 4, 5, 6]
>>> my_list.reverse()
>>> my_list
[6, 5, 4, 3, 2, 1, {1: 'one', 2: 'two', 3: 'Three'}, (1, 2, 3), [1, 2, 3], 3, 2, 1, 1]
>>> my_list.pop()
1
>>> my_list.pop()
1
>>> my_list.pop()
2
>>> my_list.pop()
3
>>> my_list
[6, 5, 4, 3, 2, 1, {1: 'one', 2: 'two', 3: 'Three'}, (1, 2, 3), [1, 2, 3]]
>>> my_list.pop()
[1, 2, 3]
>>> my_list
[6, 5, 4, 3, 2, 1, {1: 'one', 2: 'two', 3: 'Three'}, (1, 2, 3)]
>>> my_list.pop()
(1, 2, 3)
>>> my_list
[6, 5, 4, 3, 2, 1, {1: 'one', 2: 'two', 3: 'Three'}]
>>> my_list.pop()
{1: 'one', 2: 'two', 3: 'Three'}
>>> my_list.pop()
1
>>> my_list.pop()
2
>>> my_list.pop()
3
>>> my_list.pop()
4
>>> my_list.pop()
5
>>> my_list.pop()
6
>>> my_list.pop()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
IndexError: pop from empty list
>>> my_list
[]
>>> my_list.insert(0,1)
>>> my_list
[1]
>>> my_list.append(2)
>>> my_list
[1, 2]
>>> len(my_list)
2
>>> list(1,2,3)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: list() takes at most 1 argument (3 given)
>>> list([1,2,3])
[1, 2, 3]
>>> list
<class 'list'>
>>> a=list([1,2,3])
>>> a
[1, 2, 3]
>>> 

